packages/react/collapsible/src/Collapsible.tsx

```diff
@@ -152,31 +140,19 @@ type CollapsibleContentPrimitive = Polymorphic.ForwardRefComponent<
 >;
 
 const CollapsibleContent = React.forwardRef((props, forwardedRef) => {
-  const {
-    selector = getSelector(CONTENT_NAME),
-    id: idProp,
-    forceMount,
-    children,
-    ...contentProps
-  } = props;
-  const { setContentId, open, disabled } = useCollapsibleContext(CONTENT_NAME);
-  const generatedId = `collapsible-${useId()}`;
-  const id = idProp || generatedId;
-
-  React.useEffect(() => {
-    setContentId(id);
-  }, [id, setContentId]);
+  const { selector = getSelector(CONTENT_NAME), forceMount, id, children, ...contentProps } = props;
+  const context = useCollapsibleContext(CONTENT_NAME);
 
   return (
-    <Presence present={forceMount || open}>
+    <Presence present={forceMount || context.open}>
       {({ present }) => (
         <Primitive
-          data-state={getState(open)}
-          data-disabled={disabled ? '' : undefined}
+          data-state={getState(context.open)}
+          data-disabled={context.disabled ? '' : undefined}
           {...contentProps}
           selector={selector}
           ref={forwardedRef}
-          id={id}
+          id={id || context.contentId}
```

Comment:
I wasn't sure what pattern to use here?
Would you prefer I do `id={context.contentId}` and move it up just before the `{...contentProps}` to show what can be overriden (like `data-state`, `data-disabled`, etc)?

Reply:
yeah makes sense to move it up I suppose like our others :+1: